tests/ragger/boilerplate_client/boilerplate_cmd.py

```diff
@@ -0,0 +1,143 @@
+
+    def get_version(self) -> Tuple[int, int, int]:
+        try:
+            response = self.client.exchange_raw(
+                self.builder.get_version()
+            ).data
+        except ExceptionRAPDU as error:
+            raise DeviceException(error_code=error.status, ins=InsType.INS_GET_VERSION)
+
+        # response = MAJOR (1) || MINOR (1) || PATCH (1)
+        assert len(response) == 3
+
+        major, minor, patch = struct.unpack(
+            "BBB",
+            response
+        )  # type: int, int, int
+
+        return major, minor, patch
+
+    def get_app_name(self) -> str:
+        try:
+            response = self.client.exchange_raw(
+                self.builder.get_app_name()
+            ).data
+        except ExceptionRAPDU as error:
+            raise DeviceException(error_code=error.status, ins=InsType.INS_GET_APP_NAME)
+
+        return response.decode("ascii")
+
+    def get_public_key(self, bip32_path: str, display: bool = False) -> Tuple[bytes, bytes]:
+        try:
+            response = self.client.exchange_raw(
+                self.builder.get_public_key(bip32_path=bip32_path,
+                                            display=display)
+            ).data
+        except ExceptionRAPDU as error:
+            raise DeviceException(error_code=error.status, ins=InsType.INS_GET_PUBLIC_KEY)
+
+        # response = pub_key_len (1) ||
+        #            pub_key (var) ||
+        #            chain_code_len (1) ||
+        #            chain_code (var)
+        offset: int = 0
+
+        pub_key_len: int = response[offset]
+        offset += 1
+        pub_key: bytes = response[offset:offset + pub_key_len]
+        offset += pub_key_len
+        chain_code_len: int = response[offset]
+        offset += 1
+        chain_code: bytes = response[offset:offset + chain_code_len]
+        offset += chain_code_len
+
+        assert len(response) == 1 + pub_key_len + 1 + chain_code_len
+
+        return pub_key, chain_code
+
+    def sign_tx(self, bip32_path: str, transaction: Transaction) -> Tuple[int, bytes]:
+        sw: int
+        response: bytes = b""
+
+        for is_last, chunk in self.builder.sign_tx(bip32_path=bip32_path, transaction=transaction):
+            if is_last:
+                self.client.send(cla=chunk[0], ins=chunk[1],
+                                 p1=chunk[2], p2=chunk[3],
+                                 data=chunk[5:])
+                # Review Transaction
+                self.client.right_click()
+                # Displaying address
+                self.client.right_click()
+                if self.client.firmware.device == 'nanos':
+                    # due to screen size, NanoS needs 2 more screens to display the address
+                    self.client.right_click()
+                    self.client.right_click()
+                # Amount
+                self.client.right_click()
+                # Approve
+                self.client.both_click()
```

Comment:
I think that could be great to add some kind of screen content verification before each clicks.

Reply:
To be updated in another PR.